File: conseil/src/main/java/com/example/conseil/entities/FavoriteRecipe.java
package com.example.conseil.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "favorite_recipes")
public class FavoriteRecipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String recipeName;

    @Column(nullable = false)
    private int value;

    @ManyToOne
    @JoinColumn(name = "recipe_id", nullable = false)
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name = "visiteur_id", nullable = false)
    private Visiteur visiteur;

}

File: FrontEndAngular/src/app/models/favorite-recipe.ts
export class FavoriteRecipe {
  id: number;
  recipeName: string;
  value: number;
  recipeId: number; // Storing the Recipe ID (relationship to Recipe)
  visiteurId: number; // Storing the Visiteur ID (relationship to Visiteur)

  constructor(
    id: number,
    recipeName: string,
    value: number,
    recipeId: number,
    visiteurId: number
  ) {
    this.id = id;
    this.recipeName = recipeName;
    this.value = value;
    this.recipeId = recipeId;
    this.visiteurId = visiteurId;
  }
}
